tests/jax/components/training/model_updating_test.py

```diff
@@ -0,0 +1,394 @@
+            "network_agent_2"
+        ].params,
+    }
+    optstate = mock_trainer.store.opt_states
+
+    return (
+        {
+            "random_key": random_key,
+            "params": params,
+            "opt_states": optstate,
+            "batch": fake_batch,
+        },
+        mock_trainer,
+    )
+
+
+def mock_minibatch_update(carry: Tuple[Any, Any], minibatches: Any) -> Tuple[Any, Any]:
+    """mock minibatch update to test model_update_epoch in MAPGEpochUpdate class
+    Args:
+        carry: tuple include params and opt_states
+        minibatches
+    Returns:
+        carry:same carry
+        metrics: to test metrics value in test of model_update_epoch
+    """
+    metrics = {
+        "agent_0": {
+            "norm_updates": jnp.array([2, 2, 2]),
+            "norm_grad": jnp.array([2, 2, 2]),
+        },
+        "agent_1": {
+            "norm_updates": jnp.array([2, 2, 2]),
+            "norm_grad": jnp.array([2, 2, 2]),
+        },
+        "agent_2": {
+            "norm_updates": jnp.array([2, 2, 2]),
+            "norm_grad": jnp.array([2, 2, 2]),
+        },
+    }
+    return carry, metrics
+
+
+@pytest.fixture
+def mock_minibatch_update_fn() -> Callable:
+    """create mock minibatch_update function"""
+    return mock_minibatch_update
+
+
+def test_on_training_utility_fns_empty_config_optimizer(
+    mock_trainer: MockTrainer,
+) -> None:
+    """Test on_training_utility_fns from MAPGMinibatchUpdate with MAPGMinibatchUpdateConfig does not include optimizer
+
+    Args:
+        mock_trainer: Trainer
+    """
+    mini_batch_update = MAPGMinibatchUpdate()
+    mini_batch_update.config.optimizer = None
+    mini_batch_update.on_training_utility_fns(trainer=mock_trainer)
+
+    assert mock_trainer.store.optimizer != None
+    assert isinstance(mock_trainer.store.optimizer, optax.GradientTransformation)
+
+    assert list(mock_trainer.store.opt_states.keys()) == [
+        "network_agent_0",
+        "network_agent_1",
+        "network_agent_2",
+    ]
+    for net_key in mock_trainer.store.networks["networks"].keys():
+        assert mock_trainer.store.opt_states[net_key][0] == optax.EmptyState()
+        assert isinstance(
+            mock_trainer.store.opt_states[net_key][1], optax.ScaleByAdamState
+        )
+        assert mock_trainer.store.opt_states[net_key][1][0] == jnp.array([0])
+        assert list(mock_trainer.store.opt_states[net_key][1][1]) == list(
+            jax.tree_map(
+                lambda t: jnp.zeros_like(t, dtype=float),
+                mock_trainer.store.networks["networks"][net_key].params,
+            )
+        )
+        assert list(mock_trainer.store.opt_states[net_key][1][2]) == list(
+            jax.tree_map(
+                jnp.zeros_like, mock_trainer.store.networks["networks"][net_key].params
+            )
+        )
+        assert mock_trainer.store.opt_states[net_key][2] == optax.EmptyState()
+
+    assert mock_trainer.store.minibatch_update_fn
```

Comment:
Maybe also assert here that it is callable since it is supposed to be a function.